src/jenkins/InputManifest.groovy

```diff
@@ -11,25 +11,38 @@ package jenkins
 
 class InputManifest {
     class Ci implements Serializable {
-        class Image implements Serializable {
+        static class Image implements Serializable {
             String name
             String args
 
             Image(Map data) {
                 this.name = data.name
 
                 if (this.name == null) {
-                    error("Missing ci.image.name")
+                    error("Missing ci.image.<platform>.<distribution>.name")
                 }
 
                 this.args = data.args
             }
         }
 
-        Image image
+        Image image  // schema-version < 1.2
+        Map<String, Map<String, Image>> images = [:]  // schema-version >= 1.2
 
         Ci(Map data) {
-            this.image = new InputManifest.Ci.Image(data.image)
+            if (this.schemaVersion < 1.2) {
+                this.image = new InputManifest.Ci.Image(data.image)
+            }
+            else {
+                Map imageData = data.image
+                imageData.each { plat, dists ->
```

Comment:
Can you add comment on how the final map would look like for ease of understanding?

Reply:
Sure will do.